Modified code to check for whitespace

oatmealraisin opened the pull request:
Modified setters, contructors, and length checks to account for leading
and trailing whitespace.

@gabemontero @bparees ptal?

gabemontero commented on src/main/java/com/openshift/jenkins/plugins/pipeline/TimedOpenShiftBasePostAction.java:
```diff
@@ -12,8 +12,8 @@
-        this.waitTime = waitTime;
-        this.waitUnit = waitUnit;
+        this.waitTime = waitTime.trim();
+        this.waitUnit = waitUnit.trim();
```
won't jobs from older version pass in null for the waitUnit?  I seem to recall that during our reviews there.
If my recollection holds , you can get a NullPointerException.

other existing params may that have been added over time may fall into that category as well.

Hence, if the above holds, I think you should do a global replace for each field, which will look like this example using `waitUnit`:

```
waitUnit != null ? waitUnit.trim() : null;
```

gabemontero reviewed (commented):
Just one comment, though if it holds, means a pervasive change for you.

Pending that, IGTM

oatmealraisin commented:
@gabemontero I've added the check throughout, good catch, thank you! ptal?

gabemontero commented on src/main/java/com/openshift/jenkins/plugins/pipeline/ImageStreamRevisionState.java:
```diff
@@ -9,7 +9,7 @@
 		if (commitId == null)
 			throw new IllegalArgumentException("nulls not allowed");
-		this.commitId = commitId;
+		this.commitId = commitId != null ? commitId.trim() : null;
```
well, in this case, null is checked and exception thrown before hand in the line above ... we can just do commitId.trim() here.

gabemontero reviewed (commented):
Just 1 minor nit and I'm GTM.

The test job failed because of a vagrant hiccup.  We'll see if that hiccup is persistent after you push the update for the one nit.  If it persists I'll merge anyway and we can track the vagrant issue separately.

Commit:
Modified code to check for whitespace

Modified setters, contructors, and length checks to account for leading
and trailing whitespace.

oatmealraisin commented:
@gabemontero I've fixed the unnecessary check, ptal?

gabemontero commented:
IGTM ... will wait for test job to hit button.

gabemontero commented:
no vargrant hiccup this time around ... job should complete soon
